_posts/2023-11-17-transactional-outbox.md

```diff
@@ -0,0 +1,295 @@
+
+The same principle applies to many everyday business processes:
+
+* **Hotel Bookings:** When you book a hotel room online, there's often a delay before the booking is confirmed. During high-demand periods, you might even receive a notification later that the room is no longer available, prompting you to find an alternative.
+* **Online Shopping:** Shopping on platforms like Amazon involves a period where the order is processed, and only later do you get confirmation. In some cases, you might be notified of out-of-stock items or shipping delays.
+* **Flight Bookings:** Booking a flight doesn’t guarantee a seat until various checks are completed, and occasionally, overbookings lead to last-minute changes.
+
+Just as these real-world scenarios adapt and manage eventual consistency, applications in distributed systems must be designed to handle it gracefully. 
+It's essential to recognize that eventual consistency is not a flaw but a characteristic of distributed systems that, when properly managed, can lead to greater scalability and resilience. 
+Applications should be designed to expect and handle delays in data synchronization and to manage the exceptions that arise when things don't go as expected.
+
+Developers can draw lessons from these real-world analogies to better understand and implement eventual consistency in their systems. 
+By acknowledging that not all processes require or benefit from strong consistency, and by designing systems that can handle inconsistencies and adapt as needed, developers can build more robust, flexible, and scalable applications. 
+This mindset shift is crucial in successfully navigating the complexities of modern distributed systems.
+
+
+# Hands-On Guide: Implementing the Transactional Outbox Pattern with Spring Boot and RabbitMQ
+
+We will demonstrate how to implement the Transactional Outbox Pattern in a Spring Boot application, integrated with RabbitMQ for message handling. 
+For the transactional outbox mechanism, we'll utilize the [Transaction Outbox library](https://github.com/gruelbox/transaction-outbox), 
+a well-curated Java library that offers seamless integration with Spring Boot and other frameworks like Quarkus.
+
+## 1. Cloning the Repository
+
+Our example project repository is available at [SWCode Samples on GitHub](https://github.com/sw-code/swcode-samples). 
+This repository provides a "plug and play" setup with a docker-compose file that includes a PostgreSQL database and a RabbitMQ container, offering a hassle-free way to get the environment up and running.
+
+```
+git clone https://github.com/sw-code/swcode-samples.git
+cd swcode-samples/transactional-outbox
+```
+
+## Running Application
+
+1. **Starting the Application:** Use Docker to start the PostgreSQL and RabbitMQ services `docker-comose up`, 
+and then run the Spring Boot application.
+2. **Creating Users:** To test user creation and the subsequent event publishing, you can use the `/users` endpoint. 
```

Comment:
please mention the beginning of this section what the app does: apparently it inserts users? this texts feels like it was explained earlier and that I forgot it